### PR TITLE
update cosmos-sdk in v50 feature branch

### PR DESCRIPTION
**Pull request by faddat**
## Description

This PR updates the cosmos-sdk to v0.50-rc.0 and fixes a few compability issues that deal with the 
creation of validators for tests. 

### Commit Message / Changelog Entry

```bash
deps: update cosmos-sdk to v0.50-rc.0
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by faddat** [comment]
This PR is good to go for the v50 branch.

File: testing/simapp/simd/cmd/root.go [hunk 1]
```diff
@@ -205,7 +205,6 @@ func initRootCmd(rootCmd *cobra.Command, encodingConfig params.EncodingConfig, b
 
 	// add keybase, auxiliary RPC, query, genesis, and tx child commands
 	rootCmd.AddCommand(
-		rpc.StatusCommand(),
```
Comment (crodriguezvega) [inline testing/simapp/simd/cmd/root.go:208]: I see that [SDK v0.50-rc0 is doing here `server.StatusCommand()`](https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-rc.0/simapp/simd/cmd/commands.go#L129). Should we do the same?

File: testing/simapp/export.go [hunk 2]
```diff
@@ -75,7 +75,15 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
...
-		_, _ = app.DistrKeeper.WithdrawValidatorCommission(ctx, val.GetOperator())
+		valAddr, err := sdk.ValAddressFromBech32(val.GetOperator())
+		if err != nil {
+			panic(err)
+		}
+		_, err = app.DistrKeeper.WithdrawValidatorCommission(ctx, valAddr)
+		if err != nil {
+			panic(err)
+		}
+		_, _ = app.DistrKeeper.WithdrawValidatorCommission(ctx, valAddr)
```
Comment (crodriguezvega) [inline testing/simapp/export.go:86]: Is this second call to `WithdrawValidatorCommission` redundant?

**Comment by faddat** [comment]
@crodriguezvega could we merge this before:

* #4391 

I think that'll be smoother.

File: testing/simapp/export.go [hunk 3]
```diff
@@ -113,8 +121,12 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 
 	// reinitialize all validators
 	err = app.StakingKeeper.IterateValidators(ctx, func(_ int64, val stakingtypes.ValidatorI) (stop bool) {
+		valBz, err := app.StakingKeeper.ValidatorAddressCodec().StringToBytes(val.GetOperator())
```
Comment (crodriguezvega) [inline testing/simapp/export.go:124]: Maybe this is just a little bit simpler?

```suggestion
		valAddr, err := sdk.ValAddressFromBech32(val.GetOperator())
```

**Review by crodriguezvega — COMMENTED** [review]
Left a few questions for now. 🙏

**Comment by crodriguezvega** [comment]
> @crodriguezvega could we merge this before:
> 
> * [deps: update the sdk 50 branch to main #4391](https://github.com/cosmos/ibc-go/pull/4391)
> 
> I think that'll be smoother.

Sounds good to me!

File: testing/simapp/export.go (hunk 2 above)
Comment (faddat) [reply testing/simapp/export.go:86]: actually, seems yes, nice catch!

File: testing/simapp/simd/cmd/root.go (hunk 1 above)
Comment (faddat) [reply testing/simapp/simd/cmd/root.go:208]: ahhh, thank you for referencing upstream -- I'd wrongly assumed that it had joined the bundle of commands.

**Review by colin-axner — APPROVED** [review]

File: .github/workflows/golangci.yml [hunk 4]
```diff
@@ -17,10 +17,10 @@ jobs:
...
-          go-version: '1.20'
+          go-version: '1.21'
       - uses: actions/checkout@v3
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3.6.0
         with:
-          version: v1.53.3
-          args: --timeout 10m
+          version: v1.54.2
+          args: --timeout 5m
```
Comment (damiannolan) [inline .github/workflows/golangci.yml:26]: I think we bumped this timeout to 10m at some point. Not sure exactly why other than general slow timings. Let's see how it fairs at 5m again I guess.

**Review by damiannolan — APPROVED** [review]
Looks good to me 👍

**Comment by faddat** [comment]
cool!  Could we please cut a new release after this is merged?

The strangelove team made a very good point about using the notional repo for ibc-go as a replace:

* https://github.com/strangelove-ventures/interchaintest/pull/716

I think that after we cut the new release, https://github.com/strangelove-ventures/interchaintest/pull/716 can be merged and we can get back on the road with e2e.

**Comment by crodriguezvega** [comment]
> cool! Could we please cut a new release after this is merged?
> 
> The strangelove team made a very good point about using the notional repo for ibc-go as a replace:
> 
> * [sdk 50 v2 strangelove-ventures/interchaintest#716](https://github.com/strangelove-ventures/interchaintest/pull/716)
> 
> I think that after we cut the new release, [strangelove-ventures/interchaintest#716](https://github.com/strangelove-ventures/interchaintest/pull/716) can be merged and we can get back on the road with e2e.

You mean to cut a new tag from [`feat/upgrade-sdk-v0.50`](https://github.com/cosmos/ibc-go/tree/feat/upgrade-sdk-v0.50) after merging this PR? Should we merge https://github.com/cosmos/ibc-go/pull/4391https://github.com/cosmos/ibc-go/pull/4391 first though, as you proposed above?

**Comment by faddat** [comment]
actually I think that it is best that we work from exactly this place.  I feel that this spot in the dev flow is more stable.  

In order to cut from #4391 we will need to update callbacks and a few more things.  

This way, we can make a release, update interchaintest, then come back over here to finish up with the merge from main.

**Pull request merged**